server/services/accountService.go

```diff
@@ -27,29 +28,68 @@ func (service *accountService) AccountBalance(
 	_ context.Context,
 	request *types.AccountBalanceRequest,
 ) (*types.AccountBalanceResponse, *types.Error) {
-	if request.AccountIdentifier.Address == "" {
+	address := request.AccountIdentifier.Address
+	if address == "" {
 		return nil, service.errFactory.newErr(ErrInvalidAccountAddress)
 	}
 
-	accountModel, err := service.provider.GetAccount(request.AccountIdentifier.Address)
+	// The specification states:
+	// > If the currencies field is populated, only balances for the specified currencies will be returned.
+	// > If not populated, all available balances will be returned.
+	// https://www.rosetta-api.org/docs/models/AccountBalanceRequest.html
+
+	// However, we cannot fully implement this requirement at the moment.
+	var currencySymbol string
+
+	if len(request.Currencies) == 0 {
```

Comment:
More tests for this logic will be added in: https://github.com/ElrondNetwork/rosetta/pull/29.